api/web_conn.go

```diff
@@ -57,14 +60,26 @@ func (c *WebConn) readPump() {
 		return nil
 	})
 
+	if c.SessionToken == "" {
+		go func() {
+			time.Sleep(AUTH_TIMEOUT)
+			if c.SessionToken == "" {
+				l4g.Debug("websocket.read: did not authenticate in time, ip=%v", c.WebSocket.RemoteAddr())
+				c.WebSocket.Close()
+			}
+		}()
+	} else {
+		HubRegister(c)
```

Comment:
I didn't register it to the Hub since `shouldSendEvent` is going to be changed a lot and I figured it was safer if this logic was separate from that and to prevent looping through webconns you don't need.

0/5 on the ticker, it's just that tickers are usually used when you have repeating events while this is one-off.

Want to chat about it after stand-up?